python/doc_generator.py

```diff
@@ -47,13 +75,51 @@ def _get_transform_markdown(transform_yaml: Dict,
         md.h2('Parameters'),
         md.table(transform_yaml['arguments']),
         md.h2('Example'),
+        md.text(DATASET_PREVIEW_TEXT) if 'preview-data' in transform_yaml else None,
+        md.python_code(DATASET_PREVIEW_CODE) if 'preview-data' in transform_yaml else None,
+        t1,# md.text(transform_yaml['preview-data']) if 'preview-data' in transform_yaml else None,
+        md.text(TRANSFORMED_DATA_PREVIEW_TEXT) if 'preview-data' in transform_yaml else None,
         md.python_code(transform_yaml['example_code']),
+        t2,# md.text(transform_yaml['post-transform-data']) if 'preview-data' in transform_yaml else None,
         md.h2('Source Code'),
         md.github_url(transform_type_dir_name, transform_name),
         '',
     ]
-    return '\n\n'.join(markdown_elements)
+    return '\n\n'.join(filter(lambda x: x!=None, markdown_elements))
 
+def _generate_dataset_markdowns(
+    rasgo: Any,
+    dataset_id: int,
+    output_cols: List[str],
+    transform_details: Dict
+) -> List[str]:
+    """Creates markdown representation of both a pre-transform Rasgo Dataset
+    and a transformed Rasgo Dataset
+
+    Arguments:
+        :dataset_id: int: A Rasgo Community Dataset ID matching the columns named 
+        in the `example_code` section of the YAML doc
+        :output_cols: List[str]: A list of column names to include in the output
+        YAML representation
+        :transform_args: dict: TODO what do we want here? How to build the transform from args
+    """
+    ds = rasgo.get.dataset(dataset_id)
+    ds_md = ds.preview()[output_cols].to_markdown()
```

Comment:
I want to talk more about use `df.to_markdown()` vs `MarkdownTableWriter` lib, vs [`tabulate`](https://pypi.org/project/pytablewriter/)`

If we think about the user goal here, it is to make the documentation about the transform as clear as possible.

Given the tables represent datasets, and not a table of args, it may look nicer to use `tabulate` strings over Markdown tables in the docs for datasets.

Also, another thing to consider, is that we want to minimize the amount of 3rd party libraries in our open source packages. Using `df.to_makedown()` would remove the requirement of using `MarkdownTableWriter` for making the transform args table current listed as a requirement in the repo too.

Would be good to sync up with @cpdough to see which one we should move forward with. It would be helpful to see to a comparison of doing the markdown table using tabulate would look like.

Reply:
Cool. `tabulate` looks way better in text, but way worse in markdown:
![image](https://user-images.githubusercontent.com/87394236/156237858-953cf180-20e9-4728-b3b1-f53b3d6cc7bf.png)
Tabulate on top here. Looks pretty attractive.

![image](https://user-images.githubusercontent.com/87394236/156237959-262ea49e-6116-4c3f-b06a-24522386fa90.png)
Tabulate on top here, too, but it's just gibberish, while the `to_markdown()` method looks pretty nice